### PR TITLE
RHIROS-610 - OS filter info in PDF report

### PR DESCRIPTION
**Pull request by kgaikwad**

### CODE REVIEW

File: src/Components/Reports/Util.js [hunk 1]
```diff
@@ -51,11 +51,23 @@ export const generateFilterText = (filters) => {
...
+    if (hasStateFilter || hasNameFilter || hasOsFilter) {
         filterText = `\nFilters applied\n`;
         filterText = hasStateFilter ? filterText.concat(`State: ${filters.stateFilter.toString()}${filterSeparatorOnLine}`) : filterText;
         filterText = hasNameFilter ? filterText.concat(`Name: ${filters.hostnameOrId}`) : filterText;
+        if (hasOsFilter) {
+            let osText = '';
+            while (filters.osFilter.length) {
+                let osValue = filters.osFilter.splice(0, 10);
+                if (osValue) {
+                    osText = osText.concat(`${osValue.toString()}${filterSeparatorOnLine}`);
```
Comment (r14chandra) [inline src/Components/Reports/Util.js:65]: Would it be better if we have trailing spaces for values below `Operating System` ?

Screenshot from 2022-04-21 14-56-48

File: src/Components/Reports/Util.js [hunk 2]
```diff
@@ -51,11 +51,23 @@ export const generateFilterText = (filters) => {
     const filterSeparatorOnLine = '\n';
     const hasStateFilter = filters?.stateFilter?.length > 0;
     const hasNameFilter =  filters?.hostnameOrId?.length > 0;
+    const hasOsFilter =  filters?.osFilter?.length > 0;
 
-    if (hasStateFilter || hasNameFilter) {
+    if (hasStateFilter || hasNameFilter || hasOsFilter) {
         filterText = `\nFilters applied\n`;
```
Comment (r14chandra) [inline src/Components/Reports/Util.js:57]: Minor nitpick
```suggestion
        filterText = `${filterSeparatorOnLine}Filters applied${filterSeparatorOnLine}`;
```

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: src/Components/Reports/Util.js [hunk 3]
```diff
@@ -51,11 +51,13 @@ export const generateFilterText = (filters) => {
...
+    const hasOsFilter =  filters?.osFilter?.length > 0;
 
-    if (hasStateFilter || hasNameFilter) {
-        filterText = `\nFilters applied\n`;
+    if (hasStateFilter || hasNameFilter || hasOsFilter) {
+        filterText = `${filterSeparatorOnLine}Filters applied${filterSeparatorOnLine}`;
         filterText = hasStateFilter ? filterText.concat(`State: ${filters.stateFilter.toString()}${filterSeparatorOnLine}`) : filterText;
-        filterText = hasNameFilter ? filterText.concat(`Name: ${filters.hostnameOrId}`) : filterText;
+        filterText = hasNameFilter ? filterText.concat(`Name: ${filters.hostnameOrId}${filterSeparatorOnLine}`) : filterText;
+        filterText = hasOsFilter ? filterText.concat(`Operating System: ${filters.osFilter.toString()}${filterSeparatorOnLine}`) : filterText;
```
Comment (r14chandra) [inline src/Components/Reports/Util.js:60]: We can show the selected `RHEL` versions in sorted order - more readable
```suggestion
        filterText = hasOsFilter ? filterText.concat(`Operating System: ${(filters.osFilter.sort()).toString()}${filterSeparatorOnLine}`) : filterText;
```

**Review by r14chandra — CHANGES_REQUESTED** [review]

File: src/Components/Reports/Util.js (hunk 3 above)
Comment (PreetiW) [reply src/Components/Reports/Util.js:60]: @kgaikwad just a small suggestion here I feel we should move the `Name` above `State` to keep the order consistent with the systems Page UI. Let me know what you think!

**Review by PreetiW — APPROVED** [review]
Thank you for creating the PR @kgaikwad Everything looks good and working fine. 👍🏻 
I have left a small suggestion, have a look and let me know what you think!
Also can you please fix the build failure?

File: src/Components/Reports/Util.js (hunk 3 above)
Comment (kgaikwad) [reply src/Components/Reports/Util.js]: > We can show the selected RHEL versions in sorted order - more readable
Fixed!

> @kgaikwad just a small suggestion here I feel we should move the Name above State to keep the order consistent with the systems Page UI. Let me know what you think!

Moved Name above State filter while generating filter text.  Fixed existing test and added one with osFilter.

**Review by r14chandra — APPROVED** [review]
Looks good 👍🏻

**Review by PreetiW — APPROVED** [review]

**Pull request merged**